Remotion/Validation/Core/Validators/MaximumLengthValidator.cs

```diff
@@ -15,18 +15,68 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using JetBrains.Annotations;
+using Remotion.FunctionalProgramming;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
+using Remotion.Validation.Results;
 
 namespace Remotion.Validation.Validators
 {
-  public class MaximumLengthValidator : LengthValidator
+  public class MaximumLengthValidator : IMaxLengthValidator
   {
-    public MaximumLengthValidator (int max, [NotNull] ValidationMessage validationMessage)
-        : base(0, max, $"The value must have at most {max} characters.", validationMessage)
+    public int? Max { get; }
+
+    public string ErrorMessage { get; }
+
+    public ValidationMessage ValidationMessage { get; }
+
+    public MaximumLengthValidator (int? max, [NotNull] ValidationMessage validationMessage)
     {
+      ArgumentUtility.CheckNotNull(nameof(validationMessage), validationMessage);
+
+      if (max != null && max < 0)
+        throw new ArgumentOutOfRangeException("max", "Max should be larger than 0.");
```

Comment:
please create follow-up ticket (shared for all four LengthValidators) where we require the min and max-length to be greater than zero. This should be a separate task because it changes the semantics in a way that's unrelated to this current change.

Message for updated argument check in follow-up ticket: "Value must be be greater than zero."

Reply:
https://re-motion.atlassian.net/browse/RM-8851